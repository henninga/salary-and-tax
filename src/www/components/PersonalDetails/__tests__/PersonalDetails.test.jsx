import mockComponent from 'identity-obj-proxy';
import React from 'react';
import renderer from 'react-test-renderer';
import testData from '../../../../__data__/cv.json'
import { PersonalDetailsComponent } from '../index';


jest.mock('../../Ui', () => mockComponent);
jest.mock('react-gravatar', () => 'Gravatar');

describe('PersonalDetailsComponent', () => {
  it('should render', () => {
    const tree = renderer.create(<PersonalDetailsComponent data={testData} isLoaded isFetching={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should not display when loading', () => {
    const tree = renderer.create(<PersonalDetailsComponent data={testData} isFetching isLoaded={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
