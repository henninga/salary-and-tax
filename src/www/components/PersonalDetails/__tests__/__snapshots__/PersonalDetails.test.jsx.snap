// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalDetailsComponent should not display when loading 1`] = `<Loading />`;

exports[`PersonalDetailsComponent should render 1`] = `
<Section
  header="My Name - Junior POTUS"
>
  <div
    className="p-g"
  >
    <div
      className="p-g-2"
    >
      <Gravatar
        email="fredrik.oterholt@technologic.as"
        size={250}
      />
    </div>
    <div
      className="p-g-10"
    >
      <Phone
        value="48043932"
      />
      <Nationality
        value="Norsk"
      />
      <Residency
        value="Oslo"
      />
      <Email
        value="fredrik.oterholt@technologic.as"
      />
    </div>
  </div>
</Section>
`;
